docs: add server-side apply examples for updating Prometheus object (#265)
rhobs/observability-operator: docs/server-side-apply.md

> # Applying changes to Prometheus resource generated by MonitoringStack using Server Side Apply

Reviewer: ```suggestion
# Server Side Apply 

##  Customizing `Prometheus` resource generated for a MonitoringStack
```

Author: updated

  fieldsV1:
    f:metadata:
      f:labels:
        f:app.kubernetes.io/managed-by: {}
        f:app.kubernetes.io/name: {}
        f:app.kubernetes.io/part-of: {}
      f:ownerReferences:
        k:{"uid":"81da0d9a-61aa-4df3-affc-71015bcbde5a"}: {}
    f:spec:
      f:additionalScrapeConfigs: {}
      f:affinity:
        f:podAntiAffinity:
          f:requiredDuringSchedulingIgnoredDuringExecution: {}
      f:alerting:
        f:alertmanagers: {}
      f:arbitraryFSAccessThroughSMs: {}
      f:logLevel: {}
      f:podMetadata:
        f:labels:
          f:app.kubernetes.io/component: {}
          f:app.kubernetes.io/part-of: {}
      f:podMonitorSelector: {}
      f:replicas: {}
      f:resources:
        f:limits:
          f:cpu: {}
          f:memory: {}
        f:requests:
          f:cpu: {}
          f:memory: {}
      f:retention: {}
      f:ruleSelector: {}
      f:rules:
        f:alert: {}
      f:securityContext:
        f:fsGroup: {}
        f:runAsNonRoot: {}
        f:runAsUser: {}
      f:serviceAccountName: {}
      f:serviceMonitorSelector: {}
      f:thanos:
        f:baseImage: {}
        f:resources: {}
        f:version: {}
      f:tsdb: {}

>   manager: obo-demo/sample-monitoring-stack

Reviewer: shouldn't the manager be the controller or the operator and not the actual resource itself ?

Author: thats what Obo sets as the manager
https://github.com/rhobs/observability-operator/blob/a977d1554c0bf8de8578e3770a6860bc0ca244a7/pkg/reconciler/reconciler.go#L37-L38

Author: can there be more than one `MonitoringStack` managed by Obo?

Reviewer: yes, definitely but IIUC, I think we should set the fieldManager to something static that represents the obo instead of the treating it like an ownerReference.

> # Server Side Apply 

Author: moved
